library/public/utils.h

```diff
@@ -55,6 +55,18 @@ class F3D_EXPORT utils
   [[nodiscard]] static std::vector<std::string> tokenize(std::string_view str);
   // clang-format on
 
+  /**
+   * Collapse a string filesystem path by:
+   * - Expanding tilda `~` into home dir in a cross-platform way
+   * - Transform relative path into an absolute path based on basedDirectory if provided, or the
+   * current directory if not
+   * - Remove any `..` if any
+   * Rely on vtksys::SystemTools::CollapseFullPath but return empty string if the provided
+   * string is empty.
+   */
+  [[nodiscard]] static std::string collapsePath(
```

Comment:
should we use `std::filesystem::path` as arguments?

Reply:
nope, because vtksys wants std::string. I dont see the point of converting back and forth

Reply:
I agree but it's semantically more correct. And where do you plan to use this function? I suppose you will have a `fs::path` when calling `collapsePath` so you are converting to a string anyway.

Reply:
Actually I dont convert to string because I use this method on inputted string by the user, but as you see fit.